langchain/chains/combine_documents/map_rerank.py

```diff
@@ -134,6 +134,11 @@ def _process_results(
                 extra_info[key] = document.metadata[key]
         if self.return_intermediate_steps:
             extra_info["intermediate_steps"] = results
+
+        source = document.metadata.get('source')
```

Comment:
i dont think we should do this inside the chain

We return enough info for a user to do this outside the chain if they so choose

Reply:
Is there an alternate way to identify the best document based on LLM scores when map_rerank is used with ConversationalRetrievalChain?
We can find information of all the input documents (https://github.com/hwchase17/langchain/blob/master/langchain/chains/conversational_retrieval/base.py#L118), but I couldn't find a way to determine which document was used for the final answer. Information from extra_info field is not available in ConversationalRetrievalChain because run method is used.

Reply:
in L134 we can return arbitrary metadata keys from the document, so could we just extract 'source' there?